WordPress/Classes/Utility/Media/ImageDownloader.swift

```diff
@@ -53,13 +53,21 @@ class ImageDownloader {
     @discardableResult
     func downloadImage(for request: URLRequest, completion: @escaping (UIImage?, Error?) -> Void) -> ImageDownloaderTask {
         let task = session.dataTask(with: request) { (data, _, error) in
-            guard let data = data, let image = UIImage(data: data) else {
-                let error = error ?? ImageDownloaderError.failed
-                completion(nil, error)
-                return
+            guard let data = data else {
+              completion(nil, ImageDownloaderError.failed)
+              return
+            }
+            var image: UIImage?
+            if let url = request.url, url.pathExtension.lowercased() == "gif" {
+              image = RCTAnimatedImage(data: data, scale: 1)
+            } else {
+              image = UIImage.init(data: data)
+            }
+            if let finalImage = image {
+              completion(finalImage, nil)
+            } else {
+              completion(nil, ImageDownloaderError.failed)
```

Comment:
On my tests this works great!
But I was thinking if we could make it a bit more readable. What about something like:

```swift
if let gif = self.gifImage(from: request, data: data) {
    completion(gif, nil)
} else if let image = UIImage(data: data) {
    completion(image, nil)
} else {
    completion(nil, ImageDownloaderError.failed)
}
```

In this way we know right away what kind of image are we passing to the completion handler.

Reply:
Thanks for the suggestion @etoledom . I updated the code according to your suggestions, do you mind to give it another look?